### PR TITLE
Lee/implement visual3 submission functions

### PR DESCRIPTION
**Pull request by Hyukjoo-Lee**
### Description
- Form Input Validation: created a validation for the input values.
- Calendar Event : Users can create a calendar event directly from the page.
- RSVP Modal: A new RSVP modal has been added to the visual 3 wedding page.
- Link to Map Button: A link to map has been added.
- Backend Routing: Implemented a PUT route for the visual 3 component.
- Image Modal: Users can now click on an image to view it in a larger modal.

### CODE REVIEW

**Review by HanKyungSung — CHANGES_REQUESTED** [review]
Currently, there is a problem with submitting the form.

image

File: invitation-backend/src/controllers/rsvp.controller.ts [hunk 1]
```diff
@@ -45,9 +45,9 @@ export const rsvpHandler = async (req: Request, res: Response, next: NextFunctio
     } else {
       // Update the exists record.
       const updatedRecord = await rsvpModel.updateOne({
-        name: name
+          name: name
```
Comment (HanKyungSung) [inline invitation-backend/src/controllers/rsvp.controller.ts]: not sure why there is a  tab. Can you take a look?

File: invitation-backend/src/controllers/rsvp.controller.ts [hunk 2]
```diff
@@ -45,9 +45,9 @@ export const rsvpHandler = async (req: Request, res: Response, next: NextFunctio
...
       }, {
-        ...body
+          ...body
```
Comment (HanKyungSung) [inline invitation-backend/src/controllers/rsvp.controller.ts]: another tab here as well

File: invitation-backend/src/controllers/rsvp.controller.ts [hunk 3]
```diff
@@ -82,9 +82,9 @@ export const rsvpHandlerSecondVersion = async (req: Request, res: Response, next
     } else {
       // Update the exists record.
       const updatedRecord = await rsvpModel.updateOne({
-        name: name
+          name: name
```
Comment (HanKyungSung) [inline invitation-backend/src/controllers/rsvp.controller.ts]: another tab here as well

File: invitation-backend/src/controllers/rsvp.controller.ts [hunk 4]
```diff
@@ -82,9 +82,9 @@ export const rsvpHandlerSecondVersion = async (req: Request, res: Response, next
...
       }, {
-        ...body
+          ...body
```
Comment (HanKyungSung) [inline invitation-backend/src/controllers/rsvp.controller.ts]: another tab here as well

File: invitation-backend/src/controllers/rsvp.controller.ts [hunk 5]
```diff
@@ -94,11 +94,64 @@ export const rsvpHandlerSecondVersion = async (req: Request, res: Response, next
...
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const { body } = req;
+    const { name, side, menu, note, subdomain } = body;
+    // console.log(body);
+    if (
+      name === undefined ||
```
Comment (HanKyungSung) [inline invitation-backend/src/controllers/rsvp.controller.ts:106]: I think the validation for checking `participate` is missing.

**Comment by Hyukjoo-Lee** [comment]
### Description
- Solved a problem with submitting the form.
- Removed the unexpected tap in the code.
- Added a option for attendance.

**Review by HanKyungSung — CHANGES_REQUESTED** [review]
image

The form does not get reset after submit. I think we should reset the form.

**Comment by Hyukjoo-Lee** [comment]
I added reset form feature after a form submission

**Review by HanKyungSung — APPROVED** [review]

**Pull request merged**